### PR TITLE
Optimizations on inquirer and assets

### PR DESCRIPTION
**Pull request by yabirgb**
- We were using instances of the Inquirer class to call staticmethods. This uses one extra method call and a check to see if there is already an instance for the singleton that are not needed.
- If we had evmtokens already resolved and called again resolve on them we would hit again the database. This happens some times in the decoders and is not really needed

### CODE REVIEW

File: rotkehlchen/tests/unit/globaldb/test_globaldb.py [hunk 1]
```diff
@@ -1285,3 +1286,28 @@ def test_get_evm_tokens(globaldb):
     tokens = globaldb.get_evm_tokens(chain_id=ChainID.ETHEREUM, protocol=CPT_COMPOUND, exceptions=(exception_address,))  # noqa: E501
     assert len(tokens) == tokens_without_exception - 1
     assert not any(token.evm_address == exception_address for token in tokens)
+
+
+def test_asset_resolution(globaldb: GlobalDBHandler):  # pylint: disable=unused-argument
+    """Test that resolution of assets doesn't hit the database if the asset
+    has been already resolved
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/globaldb/test_globaldb.py:1293]:
```suggestion
    has already been resolved
```

**Review by LefterisJP — COMMENTED** [review]
Looks okay. Just a nitpick typo and a question. Shouldn't this go to bugfixes? As it's really small, is an improvement and would not create any problems?

**Review by LefterisJP — COMMENTED** [review]
There is some test failure heres

File: rotkehlchen/tests/unit/globaldb/test_globaldb.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from pathlib import Path
 from shutil import copyfile
 from typing import TYPE_CHECKING
+from unittest.mock import patch
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/globaldb/test_globaldb.py:6]: why is this needed here now?

**Review by LefterisJP — COMMENTED** [review]
lint failure

File: rotkehlchen/tests/unit/globaldb/test_globaldb.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/tests/unit/globaldb/test_globaldb.py]: Is me working in 3 things and having the wrong import commited. Has had to be after running the linter.sry

File: rotkehlchen/tests/unit/test_defi_oracles.py [hunk 3]
```diff
@@ -71,42 +71,43 @@ def test_uniswap_oracles_special_cases(inquirer_defi, socket_enabled):  # pylint
...
+        return original_resolver_to_class(identifier, **kwargs)
+
+    with (
+        patch.object(asset_resolver, 'resolve_asset', wraps=mocked_asset_getter),
+        patch.object(asset_resolver, 'resolve_asset_to_class', wraps=mocked_asset_getter),
+    ):
         weth = EvmToken(A_WETH.identifier)
         assert weth.decimals is None
-        with pytest.raises(DefiPoolError):
+        with pytest.raises(DefiPoolError):  # type: ignore[unreachable]
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_defi_oracles.py:110]: why is this type ignore needed?

File: rotkehlchen/inquirer.py [hunk 4]
```diff
@@ -654,8 +652,8 @@ def _find_usd_price(
             underlying_tokens = asset.underlying_tokens
 
             # Check if it is a special token
-            if asset.identifier in instance.special_tokens:
-                ethereum = instance.get_evm_manager(chain_id=ChainID.ETHEREUM)
+            if asset.identifier in Inquirer.special_tokens:
+                ethereum = Inquirer().get_evm_manager(chain_id=ChainID.ETHEREUM)
```
Comment (LefterisJP) [inline rotkehlchen/inquirer.py:656]: why do you need the parentheses in this one?

**Review by LefterisJP — COMMENTED** [review]
2 questions but all green

File: rotkehlchen/tests/unit/test_defi_oracles.py (hunk 3 above)
Comment (yabirgb) [reply rotkehlchen/tests/unit/test_defi_oracles.py]: explained in the comment

File: rotkehlchen/inquirer.py (hunk 4 above)
Comment (yabirgb) [reply rotkehlchen/inquirer.py]: Because is a method and not an staticmethod. I've changed it

File: rotkehlchen/tests/unit/test_defi_oracles.py [hunk 5]
```diff
@@ -71,42 +71,43 @@ def test_uniswap_oracles_special_cases(inquirer_defi, socket_enabled):  # pylint
...
+        return original_resolver_to_class(identifier, **kwargs)
+
+    with (
+        patch.object(asset_resolver, 'resolve_asset', wraps=mocked_asset_getter),
+        patch.object(asset_resolver, 'resolve_asset_to_class', wraps=mocked_asset_getter),
+    ):
         weth = EvmToken(A_WETH.identifier)
         assert weth.decimals is None
-        with pytest.raises(DefiPoolError):
+        with pytest.raises(DefiPoolError):  # type: ignore[unreachable]  # this is needed because decimals is not nullable in the dataclass but it is in the database and the previous line is always evaluated as False by mypy.
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_defi_oracles.py:110]: should add a link to this in the note I asked you to create. Also a link to the note from here

**Review by LefterisJP — APPROVED** [review]
LGTM
